docs/dmriprep/hmc.md

```diff
@@ -1 +1,132 @@
-# Head Motion Correction
+# Contributing to *dMRIPrep* - the case of head-motion and eddy-currents estimation
+
+All *NiPreps* are open to the community feedback and contribution.
+Collaborating to seemingly big projects (*dMRIPrep* is, in reality, pretty small) can feel scary at first.
+
+## Step 0: check the contributing guidelines
+Before contributing to an open-source initiative, your first move *should* be checking how and whether the project is seeking feedback from the outside.
+Typically, that is stated in the `CONTRIBUTING.md` file of the project or similar solution (e.g., all [*NiPreps* share contributing guidelines](https://nipreps.org/community/)).
+
+## Step 1: preparing a proposal for a new feature, some documentation, or a bugfix
+Impromptu contributions are typically difficult to adsorb by projects involving several developers and/or with many users.
+For this reason, please first navigate the project's home (https://github.com/nipreps/dmriprep).
+There are two locations where you'll find details about ongoing and future work.
+The [existing and past pull-requests tab](https://github.com/nipreps/dmriprep/pulls) will give you an idea of what new features/bugfixes are currently in progress, and you can also check the requirements to get a pull-request accepted by looking at closed (and "merged") pull-requests.
+Next to that tab, you'll typically find [the issue tracker tab](https://github.com/nipreps/dmriprep/issues/).
+There, you should check that the feature you have in mind is not already in the works.
+Opening a new issue requesting feedback and gauging the interest in a particular new idea is generally very welcome.
+
+Head to https://nipreps.org/community/features/ to learn more about the process of proposing a new feature.
+
+## Step 2: plan ahead
+Once you get the thumbs-up from the maintainers, it is time to sort out your contribution.
+
+In this book we will show how to integrate a new algorithm into *dMRIPrep*.
+First you should accurately define the problem:
+
+**Problem**: although most dMRI practitioners employ FSL for the estimation and correction of head-motion and eddy-current distortions, the FSL toolbox has restrictions of commercial use. By using FSL's eddy implementation, *dMRIPrep* inherits those restrictions.
+Therefore, a fully-open implementation of *dMRIPrep* would require eliminating FSL as a dependency.
+In this case, we will provide our own implementation of an algorithm for head-motion and eddy-current-derived distortions.
```

Comment:
I don't know if I like this framing. Arguably, we want to provide a general, open-source solution that is also best in class. The license issues, though real, seem a bit obscure.